log10/load.py

```diff
@@ -167,215 +155,142 @@ def __exit__(self, exc_type, exc_value, traceback):
+        # If session ID isn't returned, continue with degraded functionality (lost session scope).
+        #
+        if sessionID is None:
+            sessionID = get_session_id()
+
+        #
+        # Get completion ID.
+        # If we cannot get a completion id, continue with degraded functionality (no logging).
+        #
+        completion_url = url + "/api/completions"
+        r = post_request(completion_url, json={})
+        completionID = None
+        organizationSlug = None
         try:
-            with timed_block(sync_log_text + " call duration"):
-                if USE_ASYNC:
-                    threading.Thread(
-                        target=run_async_in_thread,
-                        kwargs={
-                            "completion_url": completion_url,
-                            "func": func,
-                            "result_queue": result_queue,
-                            **kwargs,
-                        },
-                    ).start()
-                else:
-                    completionID = log_sync(completion_url=completion_url, func=func, **kwargs)
-
-            current_stack_frame = traceback.extract_stack()
-            stacktrace = [
-                {
-                    "file": frame.filename,
-                    "line": frame.line,
-                    "lineno": frame.lineno,
-                    "name": frame.name,
-                }
-                for frame in current_stack_frame
-            ]
+            completionID = r.json().get("completionID")
+            organizationSlug = r.json().get("organizationSlug")
+        except Exception as e:
+            logger.warning(
+                f"LOG10: failed to get completion ID. Error: {e}. Skipping logging."
+            )
+            return func_with_backoff(func, *args, **kwargs)
+
+        url = completion_url + "/" + completionID
+
+        full_url = url + "/app/" + organizationSlug + "/completions/" + completionID
+        if DEBUG:
+            logger.debug(f"Completion URL: {full_url}")
+
+        #
+        # Create base log row (resending request in case of failure)
+        #
+        current_stack_frame = traceback.extract_stack()
+        stacktrace = [
+            {
+                "file": frame.filename,
+                "line": frame.line,
+                "lineno": frame.lineno,
+                "name": frame.name,
+            }
+            for frame in current_stack_frame
+        ]
+        log_row = {
+            "status": "started",
+            "orig_module": func.__module__,
+            "orig_qualname": func.__qualname__,
+            "stacktrace": json.dumps(stacktrace),
+            "request": json.dumps(kwargs),
+            "session_id": sessionID,
+            "tags": global_tags,
+        }
 
+        output = None
+        start_time = None
+        try:
+            #
+            # Store request
+            #
+            post_request(url, json=log_row)
```

Comment:
Maybe post request should get moved out of the try/except since it already absorbs exceptions. That way the try/except can tightly wrap the LLM call.